packages/db/index.js

```diff
@@ -503,41 +504,49 @@ export class DBClient {
    * List uploads of a given user.
    *
    * @param {number} userId
-   * @param {import('./db-client-types').ListUploadsOptions} [opts]
-   * @returns {import('./db-client-types').ListUploadReturn}
+   * @param {import('./index').PageRequest} pageRequest
+   * @returns {Promise<import('./db-client-types').ListUploadReturn>}
    */
-  async listUploads (userId, opts = {}) {
-    const rangeFrom = opts.offset || 0
-    const rangeTo = rangeFrom + (opts.size || 25)
-    const isAscendingSortOrder = opts.sortOrder === 'Asc'
-    const defaultSortByColumn = Object.keys(sortableColumnToUploadArgMap)[0]
-    const sortByColumn = Object.keys(sortableColumnToUploadArgMap).find(key => sortableColumnToUploadArgMap[key] === opts.sortBy)
-    const sortBy = sortByColumn || defaultSortByColumn
-
-    let query = this._client
-      .from('upload')
-      .select(uploadQuery, { count: 'exact' })
-      .eq('user_id', userId)
-      .is('deleted_at', null)
-      .order(
-        sortBy,
-        { ascending: isAscendingSortOrder }
-      )
-
-    // Apply filtering
-    if (opts.before) {
-      query = query.lt('inserted_at', opts.before)
-    }
-
-    if (opts.after) {
-      query = query.gte('inserted_at', opts.after)
+  async listUploads (userId, pageRequest) {
+    let query
+    if (pageRequest.type === DATE_TIME_PAGE_REQUEST) {
```

Comment:
I find the Date time pagination a little tricky,
Say I had two or more records with the exact same date time, wouldn't this break pagination & it would miss some results? 
Super edge-casey, but am keen to understand why filtering by datetime is necessary, is it purely for the existing UI?

Reply:
It's required by the pinning service API https://ipfs.github.io/pinning-services-api-spec/#section/Pagination-and-filtering and that's what was originally implemented.

Reply:
Ahh ok, in which case should this also be replicated in `listPsaPinRequests`?

Reply:
`listPsaPinRequests` serves the PSA (`GET /pins`) primarily. It's now date ordered explicitly (and you can't change the sort field or direction from the UI). So, I've left it mostly as is and dealt with the "page number pagination" at the API handler level instead. If we want to enable that UX for the PSA table then yes we can probably refactor to use this pattern also.